arch/risc-v/src/esp32c3/esp32c3_hcisocket_host.c

```diff
@@ -0,0 +1,373 @@
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#ifdef CONFIG_ESP32C3_BLE
+
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "esp32c3_ttyhci.h"
+#include "esp32c3_hcisocket_host.h"
+
+#include <time.h>
+#include <sys/time.h>
+#include "esp32c3_ble_adapter.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define BTPROTO_HCI 1
+#define HCI_CHANNEL_RAW   0
+#define HCI_CHANNEL_USER  1
+#define HCIDEVDOWN  0x400448ca
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct sockaddr_hci
+{
+  sa_family_t     hci_family;
+  unsigned short  hci_dev;
+  unsigned short  hci_channel;
+};
+
+typedef struct
+{
+    uint8_t *q_data;
+    uint16_t q_data_len;
+} host_rcv_data_t;
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static void controller_rcv_pkt_ready(void);
+static int host_rcv_pkt(uint8_t *data, uint16_t len);
+
+/****************************************************************************
+ * Public Functions declaration
+ ****************************************************************************/
+
+int bthcisock_host_avail(int fd);
+int bthcisock_host_send(int fd, const void *data, size_t len);
+int bthcisock_host_read(int fd, void *data, size_t len);
+int bthcisock_host_open(int dev_idx);
+int bthcisock_host_close(int fd);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+static int32_t hci_queue_size = 0;
+static void * hci_queue = NULL;
+static esp_vhci_host_callback_t vhci_host_cb =
+{
+    controller_rcv_pkt_ready,
+    host_rcv_pkt
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: controller_rcv_pkt_ready
+ *
+ * Description:
+ *   If the controller could receive HCI comand will callback this function.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void controller_rcv_pkt_ready(void)
+{
+}
```

Comment:
Not needed?

Reply:
Not used. but needs to be adapted.

Reply:
Maybe add some brief comment in function body mentioning why it is not needed?